src/hotspot/share/opto/castnode.cpp

```diff
@@ -96,6 +96,12 @@ Node* ConstraintCastNode::make_cast(int opcode, Node* c, Node *n, const Type *t,
     cast->set_req(0, c);
     return cast;
   }
+  case Op_CastLL: {
+    assert(!carry_dependency, "carry dependency not supported");
```

Comment:
Any particular reason to reject control dependency (except it is not used right now)?

Reply:
Because it's not used. But actually now that I think more about it, it's required. I updated the change with support for carry_dependency for CastLL.
Thanks for the review!